server/server.go

```diff
@@ -32,26 +32,41 @@ func Listen(cfg *config.Configuration, handler http.Handler, adminHandler http.H
 	adminServer := newAdminServer(cfg, adminHandler)
 	go shutdownAfterSignals(adminServer, stopAdmin, done)
 
-	mainServer := newMainServer(cfg, handler)
-	go shutdownAfterSignals(mainServer, stopMain, done)
+	if cfg.EnableSocket && len(cfg.Socket) > 0 { // start the unix_socket server if config enable-it.
+		mainServer := newSocketServer(cfg, handler)
+		go shutdownAfterSignals(mainServer, stopMain, done)
 
-	mainListener, err := newListener(mainServer.Addr, metrics)
-	if err != nil {
-		glog.Errorf("Error listening for TCP connections on %s: %v for main server", mainServer.Addr, err)
-		return
+		socketListener, err := newUnixListener(mainServer.Addr, metrics)
+		if err != nil {
+			glog.Errorf("Error listening for Unix-Socket connections on path %s: %v for socket server", mainServer.Addr, err)
+			return
+		}
+
+		go runServer(mainServer, "Socket", socketListener)
+	} else { // start the TCP server
+		mainServer := newMainServer(cfg, handler)
+		go shutdownAfterSignals(mainServer, stopMain, done)
+
+		mainListener, err := newTCPListener(mainServer.Addr, metrics)
+		if err != nil {
+			glog.Errorf("Error listening for TCP connections on %s: %v for main server", mainServer.Addr, err)
+			return
+		}
+
+		go runServer(mainServer, "Main", mainListener)
 	}
-	adminListener, err := newListener(adminServer.Addr, nil)
+
+	adminListener, err := newTCPListener(adminServer.Addr, nil)
```

Comment:
Is there a valid way to improve the test coverage for this block of code? None of these lines are currently covered.

Reply:
There might not be a good way to cover this code with unit tests. The TCP code is equally not tested, for likely the same limitations. The author attested to successful local testing.